imagepipeline/src/main/java/com/facebook/imagepipeline/producers/LocalFetchProducer.java

```diff
@@ -105,20 +105,34 @@ protected EncodedImage getByteBufferBackedEncodedImage(
     }
   }
 
-  protected EncodedImage getEncodedImage(
+    private boolean isMemoryEnough() {
+      Runtime runTime = Runtime.getRuntime();
+      long javaMax = runTime.maxMemory();
+      long javaUsed = runTime.totalMemory() - runTime.freeMemory();
+      long javaFree = Math.min(javaMax - javaUsed, 8 * ByteConstants.MB);
+      return (javaMax >= 64 * javaFree);
```

Comment:
Based on the method name, I can interfere that if the free memory is less than or equal to the maximum memory, then there is enough memory. It doesn't sound very logical.